src/filter.rs

```diff
@@ -190,18 +278,83 @@ pub(crate) fn unfilter(
                 };
             }
 
-            avg_tail!(avg_tail_8, 8);
-            avg_tail!(avg_tail_6, 6);
-            avg_tail!(avg_tail_4, 4);
-            avg_tail!(avg_tail_3, 3);
             avg_tail!(avg_tail_2, 2);
             avg_tail!(avg_tail_1, 1);
 
             match tbpp {
-                BytesPerPixel::Eight => avg_tail_8(current, previous),
-                BytesPerPixel::Six => avg_tail_6(current, previous),
-                BytesPerPixel::Four => avg_tail_4(current, previous),
-                BytesPerPixel::Three => avg_tail_3(current, previous),
+                BytesPerPixel::Eight => {
```

Comment:
Could you reverse the order of these so it matches `Sub`? We might as well make them consistent if the code is changing anyway

Reply:
Sure, I've reordered the match arms.